Marlin/Marlin_main.cpp

```diff
@@ -1429,6 +1434,108 @@ static void setup_for_endstop_move() {
   inline void do_blocking_move_to_z(float z) { do_blocking_move_to(current_position[X_AXIS], current_position[Y_AXIS], z); }
   inline void raise_z_after_probing() { do_blocking_move_to_z(current_position[Z_AXIS] + Z_RAISE_AFTER_PROBING); }
 
+  #ifdef EXT_ADC
+    /*
+     * Bed leveling probe - returns a uint16_t with ADC height value
+     */
+     static uint16_t bed_level_probe_pt(float x, float y, float z, int verbose_level=0) {
+       do_blocking_move_to(current_position[X_AXIS], current_position[Y_AXIS], current_position[Z_AXIS]);
+       do_blocking_move_to(x - 0.1, y, z);
+       sync_plan_position();
+       gcode_M241(1000);
+       bedlevelprobes[0] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS], current_position[Y_AXIS] + 0.1, z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[1] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS] + 0.1, current_position[Y_AXIS], z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[2] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS] + 0.1, current_position[Y_AXIS], z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[3] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS], current_position[Y_AXIS] - 0.1, z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[4] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS], current_position[Y_AXIS] - 0.1, z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[5] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS] - 0.1, current_position[Y_AXIS], z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[6] = gcode_M238(4);
+     
+       do_blocking_move_to(current_position[X_AXIS] - 0.1, current_position[Y_AXIS], z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[7] = gcode_M238(4);
+     
+       do_blocking_move_to(x, y, z);
+       sync_plan_position();
+       gcode_M241(350);
+       bedlevelprobes[8] = gcode_M238(4);
+     
+       uint16_t num_samples = 0x0001 << 3;
+       uint16_t i = 0;
+       uint32_t sample_sum = 0;
+       uint16_t sample_avg = 0;
+     
+       for(i = 0; i < 8; i++) {
+         sample_sum += bedlevelprobes[i];
+         if (verbose_level > 3) {
+           SERIAL_PROTOCOLPGM("sample_point #");
+           SERIAL_PROTOCOL_F(i, 10);
```

Comment:
It's to specify the base of the number. It's actually unnecessary since it defaults to 10 I believe. I'll remove.